### DISCUSSION TITLE
Converting CMYK images to RGB

### DISCUSSION THREAD

**Discussion by ColuaP** [ImageMagick/ImageMagick#6409] [Help] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows

### Description

I want to convert  hundred of images from CMYK to RGB and want to use imagemagick to do that but the result is not good.
tried several ways but the result is the same.

As you can see in the result below before and after converting, the colors are very different.
Can anyone help me?
![Screenshot 2023-06-15 at 17 39 24](https://github.com/ImageMagick/ImageMagick/assets/129497758/22857795-ff4d-43de-9a7d-d81c1eeca94a)

**Comment by snibgo** [comment]
What command did you use?

Please show the input file. The actual file, not a screenshot of some program displaying the file.

**Comment by fmw42** [comment]
Did you use profiles?  Did the input image have a profile?

**Comment by ColuaP** [comment]
I tried several commands, from the simple:  `magick CCP_flat.jpg -colorspace RGB rgb.jpg` ; to other more complex: `magick convert CCP_flat.jpg -strip -profile AdobeRGB1998.icc -colorspace RGB -strip rgb.jpg` ; but the result is very different from the original file.
The file is this one:
![CCP_flat](https://github.com/ImageMagick/ImageMagick/assets/129497758/20163472-3ba2-435a-aa70-fe8c9743e396)

**Comment by snibgo** [comment, marked as answer]
I assume you want the result to be sRGB. Is that correct?

Your file is CMYK, with no embedded ICC profile. So it contains no metadata that defines what colours are in the file. So any conversion to sRGB can only be a guess. This applies to *any* software that does the conversion, not just ImageMagick. So you don't actually *know* what the image should look like.

I suggest "`-colorspace sRGB`" (note: sRGB not RGB).
```
magick chave.jpg -colorspace sRGB out.png
```

If you don't like the result, you might guess the ICC colour profile that should be assigned. Assign that profile, and convert it to sRGB. For example:
```
magick chave.jpg -profile USWebCoatedSWOP.icc -profile sRGB.icc out2.png
```
Provide the full paths to the ICC files if they are not in your current directory.

**Reply by ColuaP** [reply]
That was the missing element.
Thanks for the help.
